security/base.py

```diff
@@ -14,17 +14,63 @@
 #    under the License.
 
 import abc
-import collections
+import datetime
 
 import six
 
 
-Issue = collections.namedtuple("Issue", ["type", "description", "subject"])
+class Issue(object):
+    __slots__ = ("type", "id", "description", "subject", "discovered_at",
+                 "confirmed_at", "resolved_at")
+
+    def __init__(self, _type, description, subject, discovered_at=None,
+                 confirmed_at=None, resolved_at=None, **kwargs):
+        self.type = _type
+        self.description = description
+        self.subject = subject
+
+        self.discovered_at = discovered_at
+        self.confirmed_at = confirmed_at
+        self.resolved_at = resolved_at
+
+        for field in ("discovered_at", "confirmed_at", "resolved_at"):
+            value = getattr(self, field)
+            if isinstance(value, datetime.datetime):
```

Comment:
And what will happen if we put here instance of something else? 

maybe we should just have something like: 

   self.discovered_at = discovered_at and discovered.isoformat("T")